### PR TITLE
docs: update after nodejs/node#4600

### PR DESCRIPTION
**Pull request by AndreasMadsen**
nodejs/node#4600 added `uid` to all hooks among other `uid` related things.

### CODE REVIEW

File: docs/AsyncWrap/example-trace/trace.js [hunk 1]
```diff
@@ -2,37 +2,45 @@
...
 
-asyncWrap.setupHooks(init, before, after);
+asyncWrap.setupHooks(init, before, after, destroy);
 asyncWrap.enable();
 
-// global state variable, that contains the current stack trace
-let currentStack = '';
+// global state variable, that contains the stack traces and the current uid
+const stack = new Map();
+      stack.set(-1, '');
```
Comment (thefourtheye) [inline docs/AsyncWrap/example-trace/trace.js]: Extra indentation

File: docs/AsyncWrap/example-trace/trace.js [hunk 2]
```diff
@@ -2,37 +2,45 @@
...
+function after(uid) {
   // At the time of writing there are some odd cases where there is no handle
   // context, this line prevents that from resulting in wrong stack trace. But
   // the stack trace will be shorter compared to what ideally should happen.
-  currentStack = '';
+  currentUid = -1;
+}
+
+function destroy(uid) {
+  // Once the handle is destroy no other handle objects can be created with
```
Comment (thefourtheye) [inline docs/AsyncWrap/example-trace/trace.js]: destroyed

File: docs/AsyncWrap/example-trace/trace.js [hunk 3]
```diff
@@ -2,37 +2,45 @@
...
+  // this handle as it immediate context. Thus its associated stack can be
```
Comment (thefourtheye) [inline docs/AsyncWrap/example-trace/trace.js]: its?

File: docs/AsyncWrap/README.md [hunk 4]
```diff
@@ -153,13 +153,13 @@ asyncWrap.disable();
 #### The Hooks
 
 Currently there are 4 hooks: `init`, `pre`, `post` `destroy`. The `this`
-variable refers to the handle object. The `init` hook has three extra arguments
-`provider`, `uid` and `parent`. The `destroy` hook also has the `uid` argument.
+variable refers to the handle object, they all have a `uid` argument, finally
+`init` provides extra information about the creation of the handle object.
```
Comment (thefourtheye) [inline docs/AsyncWrap/README.md]: this is a little vague

File: docs/AsyncWrap/README.md [hunk 5]
```diff
@@ -211,16 +211,20 @@ At the time of writing this is the current list:
 
 ##### uid
 
-The `uid` is a unique integer that identify each handle object. Because the
-`destroy` hook isn't called with the handle object, this is particular useful
-for storing information related to the handle object, that the user require in
-the `destroy` hook.
+The `uid` is a unique integer that identify each handle object. You can use
```
Comment (thefourtheye) [inline docs/AsyncWrap/README.md]: identifies

File: docs/AsyncWrap/README.md [hunk 6]
```diff
@@ -211,16 +211,20 @@ At the time of writing this is the current list:
...
+the `uid` to store information related to the handle object, by using it as
+a key for a shared `Map` object.
 
-##### parent
+As such you could also store information on the `this` object, but this is may
```
Comment (thefourtheye) [inline docs/AsyncWrap/README.md]: this may

File: docs/AsyncWrap/README.md [hunk 7]
```diff
@@ -211,16 +211,20 @@ At the time of writing this is the current list:
...
+be unsafe since the handle objects have undocumented properties. The `this`
```
Comment (thefourtheye) [inline docs/AsyncWrap/README.md]: unsafe in what sense? Do we leak information?

File: docs/AsyncWrap/example-trace/trace.js (hunk 2 above)
Comment (AndreasMadsen) [inline docs/AsyncWrap/example-trace/trace.js]: This is the name that was chosen: https://github.com/nodejs/node/blob/master/src/async-wrap.cc#L135

Comment (thefourtheye) [inline docs/AsyncWrap/example-trace/trace.js]: @AndreasMadsen I am talking about the comment :)

File: docs/AsyncWrap/README.md (hunk 4 above)
Comment (AndreasMadsen) [inline docs/AsyncWrap/README.md]: I think specifying all details is overdoing it, the signature clearly shows the details and the rest of the section is dedicated to explaining these arguments. This provides the general thought behind those extra arguments, which I actually find much more useful than just writing out their names.

File: docs/AsyncWrap/example-trace/trace.js (hunk 2 above)
Comment (AndreasMadsen) [inline docs/AsyncWrap/example-trace/trace.js]: oh, good catch :)

**Comment by AndreasMadsen** [comment]
node v5.7.0 has landed, so I'm landing this too.

**Pull request merged**